server/graphql/core/src/loader/stocktake_lines_report.rs

```diff
@@ -0,0 +1,80 @@
+use async_graphql::dataloader::*;
+use async_graphql::*;
+use repository::EqualFilter;
+use repository::{
+    RepositoryError, StocktakeLineFilter, StocktakeLineReport, StocktakeLineRepository,
+    StorageConnectionManager,
+};
+use service::report::data_sort_inputs::DataSort;
+use std::collections::HashMap;
+use std::hash::Hash;
+
+/// This loader is used to load stocktake lines by stocktake id and sort them by sort field
+/// It is used in stocktake report query
+pub struct StocktakeLineReportByStocktakeIdLoader {
+    pub connection_manager: StorageConnectionManager,
+}
+
+#[derive(Clone)]
+/// This struct is used as a key for the loader to load stocktake lines by stocktake id
+/// and sort them by sort field
+pub struct StocktakeBatchParams {
+    /// Stocktake id to load stocktake lines
+    pub stocktake_id: String,
+    /// Sort field to sort stocktake lines
+    pub sort: Option<DataSort>,
+}
```

Comment:
Workaround with passing struct as a key for loader function to pass extra arguments. Not sure if this is a best practice to follow. Would love to have a take/suggestion on this.